### PR TITLE
Upgrade all libraries

### PR DESCRIPTION
**Pull request by travishaagen**
Hey @ marcowindt, thank you for this fun app! I updated all the libraries, and replaced some libraries that are no longer supported.

This was tested on an iPhone running iOS 15. I was only able to test Android using an emulator.

Using this version of Flutter:

```
$ flutter --version
Flutter 2.12.0-4.1.pre • channel beta • https://github.com/flutter/flutter.git
Framework • revision 680962aa75 (3 weeks ago) • 2022-03-16 13:46:03 -0700
Engine • revision e9f57b5d0f
Tools • Dart 2.17.0 (build 2.17.0-182.1.beta) • DevTools 2.11.1
```

For some reason, I've had to use the following forced upgrade of a transitive dependency with these newer Flutter versions:

```
dependencies:
  platform: 3.1.0 # TRANSITIVE
```

### CODE REVIEW

File: lib/ui/screens/settings_screen.dart [hunk 1]
```diff
@@ -72,21 +69,21 @@ class _SettingsScreen extends State<SettingsScreen> {
...
-                title: Text('Invert Gryo Y'),
+                title: Text('Invert Gyro Y'),
                 trailing: Checkbox(
                   value: gyroSettings.invertGyroY,
                   onChanged: (bool value) => gyroSettings.setInvertGyroY(value),
                 ),
               ),
               ListTile(
-                title: Text('Invert Gryo Z'),
+                title: Text('Invert Gyro Z'),
```
Comment (marcowindt) [inline lib/ui/screens/settings_screen.dart:86]: Nice 🏆

**Review by marcowindt — APPROVED** [review]

**Comment by marcowindt** [comment]
Hi! Thanks for opening this PR. The app certainly needs some maintenance, thanks a lot. 

The CI check is failing due to permissions, although the check itself doesn't really do anything at the moment, but I did some testing myself and it LGTM 👍

**Comment by travishaagen** [comment]
@marcowindt happy to help! I guess you're going to merge once you get the CI checks fixed?

**Comment by marcowindt** [comment]
@travishaagen It should be fixed now, merging right away. Thanks, again!

**Pull request merged**